### PR TITLE
P20-329: Fix curriculum content i18n sync-in

### PR DESCRIPTION
**Pull request by artem-vavilov**
Because [`I18nScriptUtils.unit_directory_change?`](https://github.com/code-dot-org/code-dot-org/pull/53403/files#diff-33254942a96145eebf4363cd53e6e6d501933be5dc1590515e287557138085e5R337) checking wasn't working correctly for `curriculum_content`, we didn't rewrite already existing files, instead, we keep redacting the same already redacted file every `sync-in` run:
- Link to markdown: `https://example.org/unit1` => `<https://example.org/unit1>`
- The first redaction: `<https://example.org/unit1>` => `[https://example.org/unit1][0]`
- The next redactions: `[https://example.org/unit1][0]` => `[https://example.org/unit1][0]`

## Links
- jira ticket: [P20-329](https://codedotorg.atlassian.net/browse/P20-329)

### CODE REVIEW

**Comment by Hamms** [comment]
Again, I need to point out that `https://example.org/unit1 => <https://example.org/unit1>` is *not* a redaction; that's just standard markdown rendering. Redaction and restoration have specific meanings within the context of our i18n sync

File: bin/i18n/i18n_script_utils.rb [hunk 1]
```diff
@@ -267,9 +267,7 @@ def self.sanitize_header!(header)
   # Note we could try here to remove the old version of the file both from the
   # filesystem and from github, but it would be significantly harder to also
   # remove it from Crowdin.
-  def self.unit_directory_change?(script_i18n_name, script_i18n_filename)
-    level_content_directory = CDO.dir(File.join(I18N_SOURCE_DIR, 'course_content'))
-
+  def self.unit_directory_change?(level_content_directory, script_i18n_name, script_i18n_filename)
```
Comment (Hamms) [inline bin/i18n/i18n_script_utils.rb]: nit; probably makes more sense to just call the new argument `content_directory`; I don't think levels are particularly relevant to this logic

File: bin/i18n/resources/dashboard/curriculum_content.rb [hunk 2]
```diff
@@ -4,7 +4,7 @@ module I18n
   module Resources
     module Dashboard
       module CurriculumContent
-        I18N_SOURCE_FILE_PATH = CDO.dir(File.join(I18N_SOURCE_DIR, 'curriculum_content')).freeze
+        I18N_SOURCE_DIR_PATH = CDO.dir(File.join(I18N_SOURCE_DIR, 'curriculum_content')).freeze
```
Comment (Hamms) [inline bin/i18n/resources/dashboard/curriculum_content.rb:7]: I like the more-accurate name here!

**Review by Hamms — APPROVED** [review]
This change looks great! Thanks for digging deep into this; I think the result is a definite improvement over what we had before.

One thing I would like to see before merging is an updated comment for the `unit_directory_change?` method, which explains that we want to be able to scope this investigation to a particular category of content. Right now, the way the comment talks about subdirectories and duplicates is a bit confusing.

**Comment by artem-vavilov** [comment]
> Again, I need to point out that `https://example.org/unit1 => <https://example.org/unit1>` is _not_ a redaction; that's just standard markdown rendering. Redaction and restoration have specific meanings within the context of our i18n sync

Sorry for misleading, by `redaction` I meant [`RedactRestoreUtils.redact_data`](https://github.com/code-dot-org/code-dot-org/blob/staging/bin/i18n/resources/dashboard/curriculum_content/sync_in.rb#L63) here:
```ruby
irb(main):000:0> RedactRestoreUtils.redact_data({test_url: 'https://example.org/unit1'})
=> {"test_url"=>"<https://example.org/unit1>"}
```

**Comment by Hamms** [comment]
That method will redact things in the given string that need to be redacted, yes, but it does so by passing the whole string through a markdown renderer. The specific change `https://example.org/unit1 => <https://example.org/unit1>` is not the result of redaction, it's the result of that rendering. It's a _side effect_ of redaction, but not itself a redaction.

**Review by Hamms — APPROVED** [review]
LGTM, thanks for all the fixes!

**Comment by annaxuphoto** [comment]
> That method will redact things in the given string that need to be redacted, yes, but it does so by passing the whole string through a markdown renderer. The specific change `https://example.org/unit1 => <https://example.org/unit1>` is not the result of redaction, it's the result of that rendering. It's a _side effect_ of redaction, but not itself a redaction.

Can we briefly talk about why our redaction utils don't redact the link and instead turn it into an autolink?

**Review by annaxuphoto — APPROVED** [review]
Thanks a ton for deep diving this Artem!

**Comment by Hamms** [comment]
Because it's actually not a link on the first pass; it's just a string that happens to represent a URL. The markdown parser that we use will by default recognize strings that are URLs, and turn them into autolinks; our `link` redaction plugin will redact any link whether it's an autolink or a manual link.

```ruby
[development] dashboard > RedactRestoreUtils.redact_data({test_url: '<https://example.org/unit1>'}, ['link'])
=> {"test_url"=>"[https://example.org/unit1][0]"}
[development] dashboard > RedactRestoreUtils.redact_data({test_url: '[example](https://example.org/unit1)'}, ['link'])
=> {"test_url"=>"[example][0]"}
```

**Comment by artem-vavilov** [comment]
> Because it's actually not a link on the first pass; it's just a string that happens to represent a URL. The markdown parser that we use will by default recognize strings that are URLs, and turn them into autolinks; our `link` redaction plugin will redact any link whether it's an autolink or a manual link.
> 
> ```ruby
> [development] dashboard > RedactRestoreUtils.redact_data({test_url: '<https://example.org/unit1>'}, ['link'])
> => {"test_url"=>"[https://example.org/unit1][0]"}
> [development] dashboard > RedactRestoreUtils.redact_data({test_url: '[example](https://example.org/unit1)'}, ['link'])
> => {"test_url"=>"[example][0]"}
> ```

@annaxuphoto Plugins that we use for `curriculum_content` redaction: https://github.com/code-dot-org/code-dot-org/blob/staging/bin/i18n/resources/dashboard/curriculum_content.rb#L8
- `resourceLink`: https://github.com/code-dot-org/code-dot-org/blob/staging/bin/test/i18n/test_redact_restore_utils.rb#L73-L78
- `vocabularyDefinition`: https://github.com/code-dot-org/code-dot-org/blob/staging/bin/test/i18n/test_redact_restore_utils.rb#L80-L85

**Pull request merged**